src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/GroupFeedback/hooks.js

```diff
@@ -0,0 +1,92 @@
+import { useState, useEffect } from 'react';
+import _ from 'lodash-es';
+import messages from './messages';
+import * as module from './hooks';
+
+export const state = {
+  summary: (val) => useState(val),
+};
+
+export const groupFeedbackCardHooks = (groupFeedbacks, updateSettings) => {
+  const [summary, setSummary] = module.state.summary({ message: messages.noGroupFeedbackSummary, values: {} });
+
+  useEffect(() => {
+    if (groupFeedbacks.length === 0) {
+      setSummary({ message: messages.noGroupFeedbackSummary, values: {} });
+    } else {
+      setSummary({
+        message: messages.groupFeedbackSummary,
+        values: { groupFeedback: groupFeedbacks[0].feedback, count: (groupFeedbacks.length - 1) },
+      });
+    }
+  }, [groupFeedbacks]);
+
+  const handleAdd = () => {
+    let newId = 0;
+    if (!_.isEmpty(groupFeedbacks)) {
+      newId = Math.max(...groupFeedbacks.map(feedback => feedback.id)) + 1;
+    }
+    const groupFeedback = { id: newId, answers: [], feedback: '' };
+    const modifiedGroupFeedbacks = [...groupFeedbacks, groupFeedback];
+    updateSettings({ groupFeedbackList: modifiedGroupFeedbacks });
+  };
+
+  return {
+    summary,
+    handleAdd,
+  };
+};
+
+export const groupFeedbackRowHooks = ({ id, groupFeedbacks, updateSettings }) => {
+  // Hooks for the answers associated with a groupfeedback
+  const addSelectedAnswer = ({ value }) => {
+    const oldGroupFeedback = groupFeedbacks.find(x => x.id === id);
+    const newAnswers = [...oldGroupFeedback.answers, value];
+    const newFeedback = { ...oldGroupFeedback, answers: newAnswers };
+    const remainingFeedbacks = groupFeedbacks.filter((item) => (item.id !== id));
+    updateSettings({ groupFeedbackList: [newFeedback, ...remainingFeedbacks] });
+  };
+  const removedSelectedAnswer = ({ value }) => {
+    const oldGroupFeedback = groupFeedbacks.find(x => x.id === id);
+    const newAnswers = oldGroupFeedback.answers.filter(item => item !== value);
+    const newFeedback = { ...oldGroupFeedback, answers: newAnswers };
+    const remainingFeedbacks = groupFeedbacks.filter((item) => (item.id !== id));
+    updateSettings({ groupFeedbackList: [newFeedback, ...remainingFeedbacks] });
+  };
```

Comment:
When there are multiple rows for group feedback, the answers are not properly associated with the row. If I selected answers in the first row then select answers in the second row, the rows switch their order. The same behavior occurs when removing answers. It appears that the clicks are always being registered in the first row.

Reply:
I'll add a check to always sort the list by id at the end. Good catch.